chia/wallet/wallet_nft_store.py

```diff
@@ -0,0 +1,140 @@
+            (
+                "CREATE TABLE IF NOT EXISTS users_nfts("
+                " nft_id text PRIMARY KEY,"
+                " nft_coin_id text,"
+                " wallet_id int,"
+                " did_id text,"
+                " coin text,"
+                " lineage_proof text,"
+                " mint_height bigint,"
+                " status text,"
+                " full_puzzle blob)"
+            )
+        )
+        await self.db_connection.execute("CREATE INDEX IF NOT EXISTS nft_coin_id on users_nfts(nft_coin_id)")
+        await self.db_connection.execute("CREATE INDEX IF NOT EXISTS nft_wallet_id on users_nfts(wallet_id)")
+        await self.db_connection.execute("CREATE INDEX IF NOT EXISTS nft_did_id on users_nfts(did_id)")
+        await self.db_connection.commit()
+        return self
+
+    async def _clear_database(self) -> None:
+        cursor = await self.db_connection.execute("DELETE FROM users_nfts")
+        await cursor.close()
+        await self.db_connection.commit()
+
+    async def delete_nft(self, nft_id: bytes32, in_transaction: bool = False) -> None:
+        if not in_transaction:
+            await self.db_wrapper.lock.acquire()
+        try:
+            cursor = await self.db_connection.execute(f"DELETE FROM users_nfts where nft_id='{nft_id.hex()}'")
+            await cursor.close()
+        finally:
+            if not in_transaction:
+                await self.db_connection.commit()
+                self.db_wrapper.lock.release()
+
+    async def save_nft(
+        self, wallet_id: uint32, did_id: Optional[bytes32], nft_coin_info: NFTCoinInfo, in_transaction: bool = False
+    ) -> None:
+        if not in_transaction:
+            await self.db_wrapper.lock.acquire()
+        try:
+            cursor = await self.db_connection.execute(
+                "INSERT or REPLACE INTO users_nfts VALUES(?, ?, ?, ?, ?, ?, ?, ?, ?)",
+                (
+                    nft_coin_info.nft_id.hex(),
+                    nft_coin_info.coin.name().hex(),
+                    int(wallet_id),
+                    did_id.hex() if did_id else None,
+                    json.dumps(nft_coin_info.coin.to_json_dict()),
+                    json.dumps(nft_coin_info.lineage_proof.to_json_dict())
+                    if nft_coin_info.lineage_proof is not None
+                    else None,
+                    int(nft_coin_info.mint_height),
+                    IN_TRANSACTION_STATUS if nft_coin_info.pending_transaction else DEFAULT_STATUS,
+                    bytes(nft_coin_info.full_puzzle),
+                ),
+            )
+            await cursor.close()
+        finally:
+            if not in_transaction:
+                await self.db_connection.commit()
+                self.db_wrapper.lock.release()
+
+    async def get_nft_list(
+        self, wallet_id: Optional[uint32] = None, did_id: Optional[bytes32] = None
+    ) -> List[NFTCoinInfo]:
+        sql: str = "SELECT * from users_nfts"
```

Comment:
please don't just use `SELECT * from ...`, instead select only the columns needed.
[example of trade_store](https://github.com/Chia-Network/chia-blockchain/pull/10368#pullrequestreview-986535381)